OpenEXR/IlmImf/ImfDwaCompressor.cpp

```diff
@@ -268,8 +268,9 @@ struct DwaCompressor::Classifier
                                 " (truncated rule).");
             
         {
-            char suffix[Name::SIZE];
-            memset (suffix, 0, Name::SIZE);
+            // maximum length of string plus one byte for terminating NULL
+            char suffix[Name::SIZE+1];
```

Comment:
There'll never be more than one of these on the stack at a time, I imagine? SIZE is 256, so seems safe.

Reply:
The [Xdr::read function specialization used here](https://github.com/AcademySoftwareFoundation/openexr/blob/master/OpenEXR/IlmImf/ImfXdr.h#L868) looks like it reads a zero terminated string of maximum length 'n', not including the terminating zero, so will write at most n+1 bytes, so 256 bytes in this case.
If the read function does not encounter a zero, it will not zero terminate the 256 bytes it reads. 
The problem then occurs at the following line with the string constructor, which will attempt to read past the suffix array to find a terminating zero. By making suffix SIZE+1 and calling memset, suffix[256] is guaranteed to remain zero, so the string constructor will stop there before it reads past the array end

Reply:
The specific problem I was worried about was that stack size is a fairly esoteric thing to set for gcc, and so I also double check even relatively trivial allocs for recursion, that's where my question came from - if there was a recursion involved, 256 bytes could eat up the tiny default gcc stack quickly. I dug a little, and the apparent defaults of 1MB on msvc, and 8MB on gcc, are not in danger from this even if recursion is involved :)